rero_ils/modules/contributions/sync.py

```diff
@@ -0,0 +1,546 @@
+                    'level': verbose_level[min(verbose, 2)]
+                },
+                'file': {
+                    'class': 'logging.handlers.TimedRotatingFileHandler',
+                    'filename': os.path.join(log_dir, 'sync_mef.log'),
+                    'when': 'D',
+                    'interval': 7,
+                    'backupCount': 10,
+                    'formatter': 'standard'
+                }
+            },
+            'loggers': {
+                # for the current module
+                self.__module__: {
+                    'handlers': ['console', 'file'],
+                    'level': 'INFO',
+                    'propagate': False
+                }
+            }
+        }
+        logging.config.dictConfig(logging_config)
+        self.logger = logging.getLogger(__name__)
+        self.log_dir = log_dir
+
+    def agent_are_different(self, agent1, agent2):
+        """Check if two agent are different.
+
+        The comparison is done only on the common fields.
+
+        :param agent1: a dict representing an agent to compare.
+        :param agent2: a dict representing an agent to compare.
+        :returns: True if they are different.
+        """
+
+        def remove_fields(agent):
+            """Remove specific fields."""
+            for field in ['$schema', 'organisation', '_created', '_updated']:
+                agent.pop(field, None)
+            for source in agent['sources']:
+                for field in ['$schema', 'md5']:
+                    agent[source].pop(field, None)
+            return agent
+
+        diff = DeepDiff(
+            remove_fields(deepcopy(agent1)),
+            remove_fields(deepcopy(agent2)),
+            ignore_order=True)
+        if diff:
+            self.logger.debug(
+                f"Agent differs: {agent1['pid']}, {agent2['pid']}",
+                diff)
+            return True
+        return False
+
+    def _get_latest(self, source, pid):
+        """Query the MEF server to retrieve the last MEF for a given agent id.
+
+        :param source - string: the agent source such as idref, gnd.
+        :param pid - string: the identifier of the agent.
+        :returns: a dict of the MEF record.
+        :rtype: dictionary.
+        """
+        mef_url = current_app.config.get('RERO_ILS_MEF_AGENTS_URL')
+        url = f'{mef_url}/mef/latest/{source}:{pid}'
+        return requests.get(url).json()
+
+    def update_agents_in_document(
+        self,
+        doc_pid,
+        pids_to_replace
+    ):
+        """Updates the contribution and subjects in document.
+
+        :param doc_pid - string: document pid
+        :param source - string: agent source i.e. gnd, rero, mef
+        :param old_agent_pid - string: old agent pid from the database
+        :param new_agent_pid - string: new agent pid from the mef server
+        :param new_mef_pid - string: new MEF pid
+        :returns: true if the document has been updated
```

Comment:
It never returns other value than `True`. Is this method really need to return something ?